test/infrastructure/docker/examples/clusterclass.yaml

```
@@ -0,0 +1,118 @@
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: ClusterClass
+metadata:
+  name: clusterclass
+  namespace: default
+spec:
+  controlPlane:
+    metadata:
+    ref:
+      apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+      kind: KubeadmControlPlaneTemplate
+      name: clusterclass-control-plane
+      namespace: default
+    machineInfrastructure:
+      ref:
+        kind: DockerMachineTemplate
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+        name: "clusterclass-control-plane"
+        namespace: default
+  infrastructure:
+    ref:
+      apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+      kind: DockerClusterTemplate
+      name: clusterclass-infrastructure
+      namespace: default
+  workers:
+    machineDeployments:
+      - class: linux-worker
+        template:
+          metadata:
+          bootstrap:
+            ref:
+              kind: KubeadmConfigTemplate
+              apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+              name: "clusterclass-md-0"
+          infrastructure:
+            ref:
+              kind: DockerMachineTemplate
+              apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+              name: "clusterclass-md-1"
```

```suggestion
              name: "clusterclass-md-0"
```

Doesn't the kubeadmconfigtemplate and the dockermachionetemplate belong to the same md?

---

Understood now! I've updated the references.